gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -183,8 +185,18 @@
   }
 
   private static ImmutableSet<MapFeature> concatFeatureSets(
-      ImmutableSet<MapFeature> a, ImmutableSet<MapFeature> b) {
-    return a.<MapFeature>builder().addAll(b).build();
+      Object[] objects) {
+    ImmutableSet<MapFeature> combinedFeatureSet = ImmutableSet.<MapFeature>builder().build();
+    for (Object obj : objects) {
```

Comment:
There are much simpler and more readble solution using streams; a quick search reveals one such solution:
https://github.com/google/ground-android/pull/773/commits

You would just need to add `.map(set -> (ImmutableSet) set)` to cast items in stream. We don't need to check the cast because we're guaranteed to get `ImmutableSet`, esp. since this method is a helper used by the constructor, which has compile checks for `ImmutableSet` in the observable stream.

Reply:
updated function with 
``
stream(Arrays.asList(objects))
        .flatMap(set -> stream((ImmutableSet<MapFeature>) set)).collect(toImmutableSet());
``